src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs

```diff
@@ -169,6 +169,25 @@ public static IEnumerable<string> EnumerateRequiredProperties(string modelName,
             return propertiesList;
         }
 
+        /// <summary>
+        /// Returns the cumulative list of all discrimintor properties found in the entire model hierarchy
+        /// </summary>discrimintor
+        /// <param name="modelName">model for which to check the discrimintor properties</param>
+        /// <param name="definitions">dictionary of model definitions</param>
+        /// <param name="propertyList">List of required properties found in model hierarchy</param>
+        public static IEnumerable<string> EnumerateDiscrimintorProperties(string modelName, Dictionary<string, Schema> definitions)
+        {
+            var modelsToCheck = EnumerateModelHierarchy(modelName, definitions);
+            var propertiesList = new List<string>();
+            foreach (var modelRef in modelsToCheck)
+            {
+                if (!definitions.ContainsKey(modelRef) || definitions[modelRef].Discriminator?.Any() != true) continue;
```

Comment:
could you refactor this code?  no need to combine them in one line.

Reply:
this '!= true' can not be replaced by '!', or will compile failed